TypeConverterTest.testSQLDateConverter_whenNumberPassed fails for non-English locales

The `TypeConverterTest.testSQLDateConverter_whenNumberPassed` doesn't use a fixed locale for parsing a Date which causes problems when executed on systems with non-English locales.

```
08:59:55 [ERROR] testSQLDateConverter_whenNumberPassed(com.hazelcast.query.impl.TypeConverterTest)  Time elapsed: 0.408 s  <<< ERROR!
08:59:55 java.text.ParseException: Unparseable date: "12-December-2012"
08:59:55 	at java.text.DateFormat.parse(DateFormat.java:366)
08:59:55 	at com.hazelcast.query.impl.TypeConverterTest.testSQLDateConverter_whenNumberPassed(TypeConverterTest.java:352)
08:59:55 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
08:59:55 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
08:59:55 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
08:59:55 	at java.lang.reflect.Method.invoke(Method.java:498)
08:59:55 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
08:59:55 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
08:59:55 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
08:59:55 	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
08:59:55 	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
08:59:55 	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:98)
08:59:55 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
08:59:55 	at java.lang.Thread.run(Thread.java:748)
```

--- a/hazelcast/src/test/java/com/hazelcast/query/impl/TypeConverterTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/query/impl/TypeConverterTest.java
@@ -31,6 +31,7 @@ import java.math.BigInteger;
 import java.text.ParseException;
 import java.text.SimpleDateFormat;
 import java.util.Date;
+import java.util.Locale;
 
 import static org.hamcrest.CoreMatchers.allOf;
 import static org.hamcrest.CoreMatchers.equalTo;
@@ -348,7 +349,7 @@ public class TypeConverterTest {
 
     @Test
     public void testSQLDateConverter_whenNumberPassed() throws ParseException {
-        SimpleDateFormat formatter = new SimpleDateFormat("dd-MMM-yyyy");
+        SimpleDateFormat formatter = new SimpleDateFormat("dd-MMM-yyyy", Locale.ENGLISH);
         Date date = formatter.parse("12-December-2012");
 
         Long millis = date.getTime();

